helper/labelGenerator.js

```diff
@@ -28,8 +23,45 @@ module.exports = function( record ){
     labelParts = buildOutput(labelParts, schema[key], record);
   }
 
-  // de-dupe outputs
-  labelParts = _.unique( labelParts );
+  // de-dupe, join, trim
+  // NOTE: while it may seem odd to call `unique` on the list of label parts,
+  //  the effect is quite subtle.  Take, for instance, a result for "Lancaster, PA"
+  //  the pseudo-object is:
+  //  {
+  //    'name': 'Lancaster',
+  //    'locality': 'Lancaster',
+  //    'region_a': 'PA',
+  //    'country_a': 'USA'
+  //  }
+  //
+  //  the code up to this point generates the label:
+  //  `Lancaster, Lancaster, PA, USA`
+  //
+  //  then the `unique` call reduces this to:
+  //  `Lancaster, PA, USA`
+  //
+  //  this code doesn't have the same effect in the case of a venue or address
+  //  where the `name` field would contain the address or name of a point-of-interest
+  //
+  //  Also see https://github.com/pelias/api/issues/429 for other ways that this is bad
+  return _.unique( labelParts ).join(', ').trim();
```

Comment:
Lodash 4 something gets rid of _.unique in favor of _.uniq. Might as well update now.